uui/components/overlays/__tests__/Modals.test.tsx

```diff
@@ -165,5 +166,155 @@ describe('Modals', () => {
+                [],
+            );
+
+            return (
+                <ModalBlocker
+                    { ...props }
+                >
+                    <ModalWindow>
+                        <PickerInput
+                            dataSource={ dataSource }
+                            value={ option }
+                            onValueChange={ setOption }
+                            selectionMode="single"
+                            valueType="entity"
+                        />
+                    </ModalWindow>
+                </ModalBlocker>
+            );
+        }
+
+        function TestComponent(): ReactNode {
+            const handleOpenModalWindow = async (): Promise<void> => {
+                try {
+                    await testUuiCtx.uuiModals.show((
+                        modalWindowProps,
+                    ) => {
+                        return (
+                            <ModalWindowWithPickerInput
+                                { ...modalWindowProps }
+                                success={ successMock.mockImplementation(modalWindowProps.success) }
+                                abort={ abortMock.mockImplementation(modalWindowProps.abort) }
+                            />
+                        );
+                    });
+                } catch {}
+            };
+
+            return (
+                <>
+                    <Button
+                        caption="Open modal window"
+                        onClick={ handleOpenModalWindow }
+                    />
+                    <Modals />
+                </>
+            );
+        }
+
+        await renderWithContextAsync(
+            <TestComponent />,
+            {
+                wrapper,
+            },
+        );
+
+        const openModalWindowButton = await screen.findByRole(
+            'button',
+            {
+                name: /open modal window/i,
+            },
+        );
+        fireEvent.click(openModalWindowButton);
+        const pickerInput = await screen.findByRole('searchbox');
+        expect(pickerInput).toBeInTheDocument();
+
+        fireEvent.click(pickerInput);
+        // Checking that picker input's body is opened.
+        expect(
+            await screen.findByRole(
+                'option',
+                {
+                    name: /option 1/i,
+                },
+            ),
+        ).toBeVisible();
+
+        // First 'Escape' key press should close the picker input's options.
+        fireEvent.keyDown(
+            pickerInput,
+            {
+                key: 'Escape',
+            },
+        );
+        // Checking that picker input's body is closed.
+        expect(
+            screen.queryByRole(
+                'option',
+                {
+                    name: /option 1/i,
+                },
+            ),
+        ).not.toBeInTheDocument();
+        expect(pickerInput).toBeInTheDocument();
+        expect(successMock).not.toBeCalled();
```

Comment:
`success` is checked just to make sure, that this test will catch a regression, when it is called together with `abort` when closing the modal window (or even instead of it).